project/CodeStone1125/README.md

```diff
@@ -0,0 +1,149 @@
+# 2D Collision Detector
+
+
+## Basic Information
+
+Simulate how different type of shape collide in a 2D map.
+
+* github repository: [https://github.com/CodeStone1125/2DCollisionDetector](https://github.com/CodeStone1125/2DCollisionDetector)
+
+## Problem to Solve
+
+The problem this system wants to solve is to calculate situation for the most common shapes in 2D map like retangle, circle and Polygon.
+
+There are some algorithms can be implemented for solving this problem. Before we dive into the topic deeper, let's first talk about the the "Axis-Aligned Bounding Box" (AABB) algorithm. While it's not a single algorithm but a framework for collision detection, it forms the basis for many collision detection systems in 2D graphics and game development due to its simplicity and efficiency.
+
+* Bounding Boxes: Each object in a 2D scene is enclosed by a rectangle (a bounding box) that is aligned with the axes of the coordinate system. These rectangles are often referred to as "AABBs" because they are aligned with the axes.
+
+  However, 2D map doesn't only contain rectangle and circle, but many different Polygons. To solve the problem, objects be divided two types that's Convex Polygon and concave polygons. And Delauney Triangulation and Hertel-Mehlhorn algorithm are used respectively to improve its efficiency.
+
+* Delauney Triangulation:
```

Comment:
Delauney triangulation is a complex problem.  How much time do you want to spend on it?  And how much time do you plan to spend on the subsequent collision?

Reply:
After considering the complexity of Delaunay triangulation, I have decided to focus on the AABB algorithm, as it can handle approximate collision detection for most cases in a 2D world.